bench/src/bulk.rs

```diff
@@ -51,27 +62,72 @@ async fn server(mut incoming: quinn::Incoming) -> Result<()> {
     let quinn::NewConnection {
         mut uni_streams, ..
     } = handshake.await.context("handshake failed")?;
-    let mut stream = uni_streams
-        .next()
-        .await
-        .ok_or_else(|| anyhow!("accepting stream failed"))??;
-    trace!("stream established");
+
+    loop {
+        let mut stream = match uni_streams.next().await {
+            None => return Ok(()),
+            Some(Err(quinn::ConnectionError::ApplicationClosed(_))) => return Ok(()),
+            Some(Err(e)) => return Err(e).context("accepting stream failed"),
+            Some(Ok(stream)) => stream,
+        };
+        trace!("stream established");
+
+        let _jh: tokio::task::JoinHandle<Result<()>> = tokio::spawn(async move {
+            let mut n = 0u64;
+            while let Some((data, offset)) = stream.read_unordered().await? {
+                n = n.max(offset + data.len() as u64);
+            }
+
+            Ok(())
+        });
+    }
+}
+
+#[derive(Debug)]
+struct SendResult {
+    duration: Duration,
+    size: usize,
+    throughput: f64,
+}
+
+async fn send_data_on_stream(
```

Comment:
Can we split out a separate commit that extracts `send_data_on_stream()` out of `client()`?

Reply:
We could. But given that half of the changes are still separate (e.g. measuring durations and having a return value which contains metrics), I don't think this is a super worthwhile exercise.

Reply:
I do feel that it makes changes like this substantially easier to review, which I feel is valuable even though splitting the commit (or redoing the changes) can feel like drudgery. So while I understand your reluctance, I would appreciate it.